### PR TITLE
Treat link headers as case-insensitive

### PR DESCRIPTION
**Pull request by sisidovski**
Solve https://github.com/google/nginx-sxg-module/issues/39

Changed the header name to be lower case to do a case-insensitive comparison. Does this change make sense? Also fixed a minor typo.

### CODE REVIEW

**Comment by kumagi** [comment]
Please rebase your code to origin/master. That typo is already fixed.
And install `clang-format` and run it before push. Thank you.

File: ngx_http_sxg_filter_module.c [hunk 1]
```diff
@@ -403,9 +403,11 @@ static ngx_int_t ngx_http_sxg_header_filter(ngx_http_request_t* req) {
        part != NULL; part = part->next) {
     ngx_table_elt_t* value = part->elts;
     for (size_t i = 0; i < part->nelts; i++) {
-      if (value[i].key.len == strlen(kLinkKey) &&
-          ngx_memcmp(kLinkKey, value[i].key.data, strlen(kLinkKey)) == 0) {
-        invoke_subrequests(&value[i].value, req, ctx);
+      if (value[i].key.len == strlen(kLinkKey)) {
+        ngx_strlow(value[i].key.data, value[i].key.data, value[i].key.len);
```
Comment (kumagi) [inline ngx_http_sxg_filter_module.c:407]: Maybe, [ngx_strcasecmp()](https://github.com/nginx/nginx/blob/master/src/core/ngx_string.c#L586-L609) should be used?

Comment (sisidovski) [reply ngx_http_sxg_filter_module.c]: Done!

**Comment by sisidovski** [comment]
I think typo will be fixed in https://github.com/google/nginx-sxg-module/pull/37
Run clang-format.

**Comment by kumagi** [comment]
Thank you!

**Pull request merged**